xbmc/pvr/recordings/PVRRecordings.cpp

```diff
@@ -465,7 +465,12 @@ CPVRRecordingPtr CPVRRecordings::GetRecordingForEpgTag(const EPG::CEpgInfoTagPtr
     else
     {
       // uid is optional, so check other relevant data.
-      if (recording.second->Channel() == epgTag->ChannelTag() &&
```

Comment:
Thanks for the explanation.

2016-08-27 21:56 GMT+02:00 Kai Sommerfeld <notifications@github.com>:

> In xbmc/pvr/recordings/PVRRecordings.cpp
> <https://github.com/xbmc/xbmc/pull/10352#discussion_r76521636>:
>
> > @@ -465,7 +465,12 @@ CPVRRecordingPtr CPVRRecordings::GetRecordingForEpgTag(const EPG::CEpgInfoTagPtr
> >      else
> >      {
> >        // uid is optional, so check other relevant data.
> > -      if (recording.second->Channel() == epgTag->ChannelTag() &&
>
> @fritsch <https://github.com/fritsch> Although the diff looks like it at
> the first glance, the fix is not circumventing a deadlock caused by
> CPVRRecording's lock by now calling CPVRRecording methods which are not
> protected by a lock. CPVRRecording's mutex is not involved in the deadlock.
>
> It's a deadlock caused by CEpg's instance lock and CPVRChannelGroup's lock.
>
> Here are the involved callstacks and a short explanation of the fix idea:
> http://pastebin.com/9u6D0VdP
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/xbmc/xbmc/pull/10352/files/5377f64ecda99f58d0a145b86605b140cb592e32#r76521636>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABCfHYaaWbaRuRhjQvmftec1K5PwJebyks5qkJZZgaJpZM4Juucr>
> .
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B